src/iceberg_manifest.cpp

```diff
@@ -1,72 +1,166 @@
+	//// lower_bounds: map<126: int, 127: binary> - 125
+	// children.emplace_back("lower_bounds", LogicalType::MAP(LogicalType::INTEGER, LogicalType::BLOB));
+	//// upper_bounds: map<129: int, 130: binary> - 128
+	// children.emplace_back("upper_bounds", LogicalType::MAP(LogicalType::INTEGER, LogicalType::BLOB));
+
+	// data_file: struct(...) - 2
+	names.push_back("data_file");
+	types.push_back(LogicalType::STRUCT(std::move(children)));
+	data_file_field_ids.emplace_back("__duckdb_field_id", Value::INTEGER(DATA_FILE));
+	field_ids.emplace_back("data_file", Value::STRUCT(data_file_field_ids));
+
+	//! Populate the DataChunk with the data files
+
+	DataChunk data;
+	data.Initialize(allocator, types, manifest_file.data_files.size());
+
+	for (idx_t i = 0; i < manifest_file.data_files.size(); i++) {
+		auto &data_file = manifest_file.data_files[i];
+		idx_t col_idx = 0;
+
+		//! We rely on inheriting the snapshot_id, this is only acceptable for ADDED data files
+		D_ASSERT(data_file.status == IcebergManifestEntryStatusType::ADDED);
+
+		// status: int - 0
+		data.SetValue(col_idx++, i, Value::INTEGER(static_cast<int32_t>(data_file.status)));
+		// snapshot_id: long - 1
+		data.SetValue(col_idx++, i, Value(LogicalType::BIGINT));
+		// sequence_number: long - 3
+		data.SetValue(col_idx++, i, Value(LogicalType::BIGINT));
+		// file_sequence_number: long - 4
+		data.SetValue(col_idx++, i, Value(LogicalType::BIGINT));
+		// data_file: struct(...) - 2
+		data.SetValue(col_idx, i, data_file.ToDataFileStruct(data.data[col_idx].GetType()));
+		col_idx++;
 	}
-	return true;
-}
+	data.SetCardinality(manifest_file.data_files.size());
 
-void AvroScan::InitializeChunk(DataChunk &chunk) {
-	chunk.Initialize(context, return_types, STANDARD_VECTOR_SIZE);
-}
+	CopyInfo copy_info;
+	copy_info.is_from = false;
+	copy_info.options["root_name"].push_back(Value("manifest_entry"));
+	copy_info.options["field_ids"].push_back(Value::STRUCT(field_ids));
+
+	CopyFunctionBindInput input(copy_info);
+	input.file_extension = "avro";
+
+	ThreadContext thread_context(context);
+	ExecutionContext execution_context(context, thread_context, nullptr);
+	auto bind_data = copy.copy_to_bind(context, input, names, types);
+
+	auto global_state = copy.copy_to_initialize_global(context, *bind_data, manifest_file.path);
+	auto local_state = copy.copy_to_initialize_local(execution_context, *bind_data);
 
-bool AvroScan::Finished() const {
-	return finished;
+	copy.copy_to_sink(execution_context, *bind_data, *global_state, *local_state, data);
+	copy.copy_to_combine(execution_context, *bind_data, *global_state, *local_state);
+	copy.copy_to_finalize(context, *bind_data, *global_state);
+
+	// auto &file_system = FileSystem::GetFileSystem(db);
+	// auto file_handle = file_system.OpenFile(manifest_file.path, FileOpenFlags::FILE_FLAGS_READ);
+	// auto manifest_length = file_handle->GetFileSize();
+	// return manifest_length;
+	return 42069;
```

Comment:
For some reason I wasn't able to OpenFile the file that was just created, which *did* work before, I believe?